emg/emg_local_STFT_execute.py

```diff
@@ -0,0 +1,172 @@
+# Runs a local BSA analysis (see emg_local_BSA.py) on one trial of EMG data. Runs on the HPC
+
+############################################################
+# Imports
+############################################################
+
+# Import stuff
+import numpy as np
+import easygui
+import os
+import sys
+import datetime
+import scipy
+import json
+
+class Logger(object):
+    def __init__(self, log_file_path):
+        self.terminal = sys.stdout
+        self.log = open(log_file_path, "a")
+
+    def append_time(self, message):
+        now = str(datetime.datetime.now())
+        ap_msg = f'[{now}] {message}'
+        return ap_msg
+
+    def write(self, message):
+        ap_msg = self.append_time(message)
+        self.terminal.write(ap_msg)
+        self.log.write(ap_msg)  
+
+    def flush(self):
+        self.terminal.flush()
+        self.log.flush()  
+
+############################################################
+# Define functions 
+############################################################
+
+def calc_stft(trial, max_freq,time_range_tuple,\
+            Fs,signal_window,window_overlap):
+    """
+    trial : 1D array
+    max_freq : where to lob off the transform
+    time_range_tuple : (start,end) in seconds, time_lims of spectrogram
+                            from start of trial snippet`
+    """
+    f,t,this_stft = scipy.signal.stft(
+                scipy.signal.detrend(trial),
+                fs=Fs,
+                window='hann',
```

Comment:
Might cause versioning issues with scipy. Keep an eye on.

Reply:
commented out to avoid versioning issues